src/azdata.proposed.d.ts

```diff
@@ -397,9 +386,23 @@ declare module 'azdata' {
 	}
 
 	export interface ConnectionProvider extends DataProvider {
+		/**
+		 * Changes the password for a connection profile.
```

Comment:
please state that this is for the password expiration scenario to avoid confusion.

Reply:
Extensions can get providers directly using azdata.dataprotocol.getProvider - and thus call changePassword themselves. So I don't think we should have implementation-specific comments like that (but also see my other comments about whether we should have these public APIs at all)